plugins/badges-backend/src/lib/BadgeBuilder/DefaultBadgeBuilder.test.ts

```diff
@@ -101,5 +112,25 @@ describe('DefaultBadgeBuilder', () => {
         markdown: `![unknown badge: other-id](${context.badgeUrl})`,
       });
     });
+
+    it('throws InputError when the BadgeFactory.createBadge() fails', async () => {
+      expect.assertions(1);
```

Comment:
Is this needed?

Reply:
maybe not.. I'm not that familiar with typescript, and what level of runtime guarantees you get. It is not the test itself, but rather the code it tests that you question, I gather.. 

https://github.com/backstage/backstage/blob/dcb5d5e373f8b300389a248b18d2beb831660d21/plugins/badges-backend/src/lib/BadgeBuilder/DefaultBadgeBuilder.ts#L39-L43

So perhaps that `if` could just be dropped altogether.

Reply:
Nah not needed in this case, it's just making sure that one assertion was made, but the assertion below is made in-line with the test anyway, so will always pass/fail.